backend/tests/security/test_materialized_views_security.py

```
@@ -0,0 +1,661 @@
+
+    @pytest.mark.asyncio
+    async def test_get_row_count_rejects_malformed_names(self):
+        """Test that get_row_count rejects malformed view names"""
+        from src.services.materialized_views import MaterializedViewRefreshService
+
+        # Setup
+        mock_db = AsyncMock()
+        service = MaterializedViewRefreshService(mock_db)
+
+        # Execute
+        with pytest.raises(ValueError, match="(Unknown materialized view|Invalid SQL identifier)"):
```

Ruff (0.14.3)

Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)